src/main/java/io/github/nucleuspowered/nucleus/modules/core/commands/MigrateCommand.java

```diff
@@ -0,0 +1,130 @@
+import io.github.nucleuspowered.nucleus.modules.jail.handlers.JailHandler;
+import io.github.nucleuspowered.nucleus.modules.mute.handler.MuteHandler;
+import ninja.leaping.configurate.objectmapping.ObjectMappingException;
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.command.CommandResult;
+import org.spongepowered.api.command.CommandSource;
+import org.spongepowered.api.command.args.CommandContext;
+import org.spongepowered.api.entity.Transform;
+import org.spongepowered.api.item.ItemType;
+import org.spongepowered.api.item.ItemTypes;
+import org.spongepowered.api.world.Location;
+import org.spongepowered.api.world.World;
+
+import java.io.IOException;
+import java.util.UUID;
+
+/**
+ * Saves the data files.
+ *
+ * Permission: nucleus.nucleus.migrate
+ */
+@RunAsync
+@NoCooldown
+@NoCost
+@NoWarmup
+@Permissions(root = "nucleus")
+@RegisterCommand(value = "migrate", subcommandOf = NucleusCommand.class)
+public class MigrateCommand extends CommandBase<CommandSource> {
+
+    @Inject private UserConfigLoader userConfigLoader;
+    @Inject private JailHandler jailHandler;
+    @Inject private MuteHandler muteHandler;
+
+    @Override
+    public CommandResult executeCommand(CommandSource src, CommandContext args) throws Exception {
+        if (Sponge.getPluginManager().getPlugin("io.github.hsyyid.essentialcmds").isPresent()) {
+            // Warps
+            for (Object warp : Utils.getWarps()) {
+                String warpName = String.valueOf(warp);
+                Transform<World> warpLocation = Utils.getWarp(warpName);
+
+                if (warpLocation != null) {
+                    NucleusWarpService qs = Sponge.getServiceManager().provideUnchecked(NucleusWarpService.class);
+                    qs.setWarp(warpName, warpLocation.getLocation(), warpLocation.getRotation());
+                }
+            }
+
+            // Blacklisted items
+            for (String item : Utils.getBlacklistItems()) {
+                ItemType itemType = Sponge.getRegistry().getType(ItemType.class, item).orElse(ItemTypes.NONE);
+
+                if (itemType != ItemTypes.NONE) {
+                    GeneralDataStore dataStore = this.plugin.getGeneralDataStore();
+                    dataStore.addBlacklistedType(itemType);
+                }
+            }
+
+            // Homes
+            Configurable homesConfig = HomeConfig.getConfig();
+
+            for (Object uuid : Configs.getConfig(homesConfig).getNode("home", "users").getChildrenMap().keySet()) {
+                UUID uniqueId = UUID.fromString(String.valueOf(uuid));
+
+                for (Object home : Utils.getHomes(uniqueId)) {
+                    String homeName = String.valueOf(home);
+                    Transform<World> homeLocation = Utils.getHome(uniqueId, homeName);
+                    InternalNucleusUser iqsu = plugin.getUserLoader().getUser(uniqueId);
+                    iqsu.setHome(homeName, homeLocation.getLocation(), homeLocation.getRotation());
```

Comment:
Question: what is your default home name? Nucleus' is `home`. If yours isn't, is it worth converting it?